protocols/mdns/src/iface.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
```

Comment:
Should this file not rather live in `src/behaviour/iface.rs`? It is only used within `src/behaviour.rs`, right?

Reply:
not sure how adding more directories for a small crate helps. is there some "policy" for the rust-libp2p repo somewhere?

Reply:
I am following the convention: If module B is used in module A only, move it into `a/b.rs`. I think we are doing a decent job following this convention.

Reply:
well, I don't like large nested repos. some directories make sense sometimes when organizing files, but if a crate has 4 files, I don't think it's helpful. Made the change (still open to reverting if you change your mind)